controller/state.go

```diff
@@ -194,6 +195,34 @@ func (m *appStateManager) GetRepoObjs(app *v1alpha1.Application, sources []v1alp
 			return nil, nil, fmt.Errorf("failed to get Kustomize options for source %d of %d: %w", i+1, len(sources), err)
 		}
 
+		val, ok := app.Annotations[v1alpha1.AnnotationKeyManifestGeneratePaths]
+		if !source.IsHelm() && app.Status.Sync.Revision != "" && ok && val != "" {
```

Comment:
for multiple source app, `app.Status.Sync.Revision != ""` is not going to be valid. It needs to check with `app.Status.Sync.Revisions[i]`

Reply:
Done